GDJS/Runtime/gd.ts

```diff
@@ -74,24 +77,54 @@ namespace gdjs {
       : [0, 0, 0];
   };
 
+  /**
+   * Convert a shorthand Hex string (#1F4) to an RGB color array [r, g, b], where each component is in the range [0, 255].
+   *
+   * @param {string} hex Color hexadecimal
+   */
+  export const shorthandHexToRGBColor = function (
+    hexString: string
+  ): [number, number, number] {
+    const hexNumber = parseInt(hexString.replace('#', ''), 16);
+    return Number.isFinite(hexNumber)
+      ? [
+          17 * ((hexNumber >> 8) & 0xf),
+          17 * ((hexNumber >> 4) & 0xf),
+          17 * (hexNumber & 0xf),
+        ]
+      : [0, 0, 0];
+  };
+
   /**
    * Convert a RGB string ("rrr;ggg;bbb") or a Hex string ("#rrggbb") to a RGB color array ([r,g,b] with each component going from 0 to 255).
    * @param value The color as a RGB string or Hex string
    */
   export const rgbOrHexToRGBColor = function (
     value: string
   ): [number, number, number] {
-    const splitValue = value.split(';');
-    // If a RGB string is provided, return the RGB object.
-    if (splitValue.length === 3) {
-      return [
-        parseInt(splitValue[0], 10),
-        parseInt(splitValue[1], 10),
-        parseInt(splitValue[2], 10),
-      ];
+    const rgbColor = extractRGBString(value);
+    if (rgbColor) {
+      const splitValue = rgbColor.split(';');
+      // If a RGB string is provided, return the RGB object.
+      if (splitValue.length === 3) {
+        return [
+          Math.min(255, parseInt(splitValue[0], 10)),
+          Math.min(255, parseInt(splitValue[1], 10)),
+          Math.min(255, parseInt(splitValue[2], 10)),
+        ];
+      }
+    }
+
+    const hexColor = extractHexString(value);
+    if (hexColor) {
+      return hexToRGBColor(hexColor);
+    }
+    const shorthandHexColor = extractShorthandHexString(value);
+    if (shorthandHexColor) {
+      return shorthandHexToRGBColor(shorthandHexColor);
     }
-    // Otherwise, convert the Hex to RGB.
-    return hexToRGBColor(value);
+    console.warn(`Could not use string ${value} as a color.`);
```

Comment:
I would not warn because this risks polluting the console with tons of fairly "low impact" logs.

Reply:
Removed ✅